### PR TITLE
skip button

### PR DESCRIPTION
**Pull request by n4b3l**
very basic implementation, I was not sure how skinable objects will work so used a regular button for now.

### CODE REVIEW

**Comment by peppy** [comment]
all of this logic should be in its own class (`SkipButton : Button`). Don't pollute `GameMode`s with things like this.

File: osu.Game/Graphics/UserInterface/SkipButton.cs [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+            sourceClock = clock;
+            this.time = time;
+        }
+
+        protected override void LoadComplete()
+        {
+            base.LoadComplete();
+
+            Delay(time - 3000, true);
+            Content.FadeOut(250);
```
Comment (peppy) [inline osu.Game/Graphics/UserInterface/SkipButton.cs]: why content.fadeout here?

File: osu.Game/Graphics/UserInterface/SkipButton.cs [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+        }
+
+        private void skip()
+        {
+            sourceClock.Seek(time - 3000);
+            FadeOut(250);
```
Comment (peppy) [inline osu.Game/Graphics/UserInterface/SkipButton.cs]: why fadeout here? is this even needed when you already have the fadeout in loadcomplete?

File: osu.Game/Graphics/UserInterface/SkipButton.cs [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+using osu.Framework.Graphics.UserInterface;
+using osu.Framework.Input;
+using osu.Framework.Timing;
+
+namespace osu.Game.Graphics.UserInterface
+{
+    class SkipButton : Button
+    {
+        private IAdjustableClock sourceClock;
+        private double time;
```
Comment (peppy) [inline osu.Game/Graphics/UserInterface/SkipButton.cs]: something more descriptive than `time` please.

**Comment by peppy** [comment]
Just wanted to prod this one, as it wasn't in such a bad state if you address the changes I mentioned.

File: osu.Game/Graphics/UserInterface/SkipButton.cs (hunk 2 above)
Comment (ZLima12) [inline osu.Game/Graphics/UserInterface/SkipButton.cs]: @peppy if I understand correctly, if not faded out here, the button would not fade out until three seconds before the first hit object.

Comment (peppy) [inline osu.Game/Graphics/UserInterface/SkipButton.cs]: no

**Comment by peppy** [comment]
No progress here, so I rewrote it at https://github.com/ppy/osu/pull/298.

**Pull request closed without merging**